src/commands/publish.ts

```diff
@@ -312,58 +327,82 @@ async function checkRevisionStatus(
   await statusProvider.waitForTheBuildToSucceed(revision);
 }
 
+/**
+ * Determines the closest branch we can merge to from the current checkout
+ * Adapted from https://stackoverflow.com/a/55238339/90297
+ * @param git our local Git client
+ * @param remoteName Name of the remote to query for the default branch
+ * @returns
+ */
+async function getMergeTarget(
+  git: SimpleGit,
+  remoteName: string
+): Promise<string> {
+  const logOutput = await git.raw(
+    'log',
+    '--decorate',
+    '--simplify-by-decoration',
+    '--oneline'
+  );
+  logger.debug('Trying to find merge target:');
+  logger.debug(logOutput);
+  const branchName =
+    stripRemoteName(
+      logOutput
+        .match(/^[\da-f]+ \((?!HEAD )([^)]+)\)/m)?.[1]
+        ?.split(',', 1)?.[0],
+      remoteName
+    ) || (await getDefaultBranch(git, remoteName));
+
+  if (!branchName) {
+    throw new Error('Cannot determine where to merge to!');
+  }
+
+  return branchName;
+}
+
 /**
  * Deals with the release branch after publishing is done
  *
  * Leave the release branch unmerged, or merge it but not delete it if the
  * corresponding flags are set.
  *
- * @param github Github client
- * @param githubConfig Github repository configuration
- * @param branchName Release branch name
- * @param skipMerge If set to "true", the branch will not be merged
+ * @param git Git client
+ * @param remoteName The git remote name to interact with
+ * @param branch Name of the release branch
+ * @param [mergeTarget] Branch name to merge the release branch into
  * @param keepBranch If set to "true", the branch will not be deleted
  */
 async function handleReleaseBranch(
-  github: Github,
-  githubConfig: GithubGlobalConfig,
-  branchName: string,
-  skipMerge = false,
+  git: SimpleGit,
+  remoteName: string,
+  branch: string,
+  mergeTarget?: string,
   keepBranch = false
 ): Promise<void> {
-  if (!branchName || skipMerge) {
-    logger.info('Skipping the merge step.');
-    return;
+  if (!mergeTarget) {
+    mergeTarget = await getMergeTarget(git, remoteName);
   }
+  logger.debug(`Checking out merge target branch:`, mergeTarget);
+  await git.checkout(mergeTarget);
```

Comment:
Does this function run _after_ dirty checks on the repository? In rare cases where we still run publish locally, it would be good to warn the same way as prepare does.

Reply:
Don't think we need dirty checks as `git checkout` itself fails when the working directory is not clean. That's what I relied on. If you think this is dangerous, I'll follow up with explicit checks tho.

Reply:
You're right, the checkout will bail in time. Thanks for clarifying 👍